shortfin/tests/apps/sd/serving_test.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import requests
+import time
+import pytest
+import subprocess
+import os
+import socket
+import sys
+import copy
+from contextlib import closing
+import numpy
+from PIL import Image
+
+from shortfin_apps.types.Base64CharacterEncodedByteSequence import (
+    Base64CharacterEncodedByteSequence,
+)
+
+from shortfin_apps.utilities.image import (
+    image_from,
+)
+
+BATCH_SIZES = [1]
+
+sample_request = {
+    "prompt": [
+        " a cat under the snow with blue eyes, covered by snow, cinematic style, medium shot, professional photo, animal",
+    ],
+    "neg_prompt": ["Watermark, blurry, oversaturated, low resolution, pollution"],
+    "height": [1024],
+    "width": [1024],
+    "steps": [20],
+    "guidance_scale": [7.5],
+    "seed": [0],
+    "output_type": ["base64"],
+    "rid": ["string"],
+}
+
+
+def start_server(fibers_per_device=1, isolation="per_fiber"):
+    # Start the server
+    srv_args = [
+        "python",
+        "-m",
+        "shortfin_apps.sd.server",
+    ]
+    with open("sdxl_config_i8.json", "wb") as f:
+        r = requests.get(
+            "https://sharkpublic.blob.core.windows.net/sharkpublic/sdxl/04082025/configs/sdxl_config_i8.json",
+            allow_redirects=True,
+        )
+        f.write(r.content)
+    srv_args.extend(
+        [
+            f"--model_config=sdxl_config_i8.json",
+            f"--fibers_per_device={fibers_per_device}",
+            f"--isolation={isolation}"
+        ]
+    )
+    runner = ServerRunner(srv_args)
+    # Wait for server to start
+    time.sleep(3)
+    return runner
+
+
+@pytest.fixture(scope="module")
+def sd_server_fpd1():
+    runner = start_server(fibers_per_device=1)
+
+    yield runner
+
+    if os.path.exists("sdxl_config_i8.json"):
+        os.remove("sdxl_config_i8.json")
+
+    # Teardown: kill the server
+    del runner
+
+@pytest.mark.system("amdgpu")
+def test_sd_server(sd_server_fpd1):
+    imgs, status_code = send_json_file(sd_server_fpd1.url)
+    assert len(imgs) == 1
+    ref_path = os.path.dirname(__file__)
+    ref_path = os.path.join(ref_path, "refs/snow_cat.png")
+    if not os.path.exists(ref_path):
+        assert 0, f"Reference image does not exist at {ref_path}"
+    ref_image = Image.open(ref_path)
+    ref_image_numpy = numpy.array(ref_image)
+    gen_image_numpy = numpy.array(imgs[0])
+    assert (ref_image_numpy == gen_image_numpy).all()
```

Comment:
Exact image comparison against a compressed image? How reliable is that? Shouldn't this use a perplexity check or some other ML-friendly metric?